File: Python Course/Software Development Module - in Python/Week13/Temperature.py
"""
Temperature.py
Billy Cussen
19/10/2020
"""

class myTemperature:
    def __init__(self, weeks,temperatures):
        self.weeks = weeks
        self.temperatures = temperatures
        self.result = "***Average Temperatures***"

    def getAverages(self):

        self.result += "\nWeek 1: "+str(sum(self.temperatures[0])/7)
        if self.weeks >= 2:
            self.result += "\nWeek 2: "+str(sum(self.temperatures[1])/7)
            if self.weeks == 3:
                self.result += "\nWeek 3: "+str(sum(self.temperatures[2])/7)
        print(self.result)

File: Java/Software Development Module - Java/Week13/Temperature.java
/*
Temperature.java
Billy Cussen
27/07/2020
*/

public class Temperature{

    //Data Members
    private String[] temperatureWeek;
    private String result;

    //Constructor
    public Temperature(){

    }

    public Temperature(String[] temperatureWeek){
        this.temperatureWeek = temperatureWeek;
        this.result = new String();
    }

    public String getResult(){

        int week1Average = 0;
        int week2Average = 0;
        int week3Average = 0;
        String[] weekCheck;

        for(int i = 0; i < this.temperatureWeek.length; i++){
            weekCheck = this.temperatureWeek[i].split(",");

            for(String day : weekCheck){

                if(i+1 == 1){
                    week1Average+= Integer.parseInt(day);
                } else if(i + 1 == 2){
                    week2Average+= Integer.parseInt(day);
                } else{
                    week3Average+= Integer.parseInt(day);
                }

            }

        }

        weekCheck = null;

        week1Average = week1Average / 7;
        //System.out.println("Week 1: "+ week1Average);
        week2Average = week2Average / 7;
        //System.out.println("Week 2: "+ week2Average);
        week3Average = week3Average / 7;
        //System.out.println("Week 3: "+ week3Average);

        this.result = "Week 1 Average: "+week1Average+"\nWeek 2 Average: "+week2Average+"\nWeek 3 Average: "+week3Average;
        this.temperatureWeek = null;

        return this.result;
    }

}